astropy/stats/funcs.py

```diff
@@ -759,22 +768,43 @@ def median_absolute_deviation(a, axis=None):
     # See https://github.com/numpy/numpy/issues/7330 why using np.ma.median
     # for normal arrays should not be done (summary: np.ma.median always
     # returns an masked array even if the result should be scalar). (#4658)
-    if isinstance(a, np.ma.MaskedArray):
+    if isinstance(data, np.ma.MaskedArray):
         func = np.ma.median
+        if ignore_nan:
+            data = np.ma.masked_where(np.isnan(data), data)
+    elif ignore_nan:
+        if NUMPY_LT_1_10 or axis is None:
+            # older versions don't have nanmedian, but nanmedian
+            # may be faster than np.ma.median *if* axis is None
+            # https://gist.github.com/MSeifert04/1551e80e1aa85b48c9191e727c38f9a9
```

Comment:
I think that's what this says.  However, the comment is right and the code is wrong...